### PR TITLE
one_source.py

### PR DESCRIPTION
**Pull request by dneise**
In this PR I try to make one_source.py working. 

I know Sebastian wants to use amalgamate rather than a homebrewed solution, but I think it is great already the way it is ... and we can still use amalgamate, if it is better.

### CODE REVIEW

**Comment by dneise** [comment]
Current state is: There is still a mistake in `merlict.h` where two classes are not defined in the right place. I am looking into it now .. 

but when I manually move these two class defininitions a little bit up .. it works like a charm!!!

**building**
```
(base) ✔ ~/merlict_development_kit [tidy_up_one_source_py|✚ 1…1] 
11:18 $ time g++ merlict_test.cpp merlict.cpp -o merlict_test

real	0m15.018s
user	0m14.513s
sys	0m0.504s
```

**testing**
```
(base) ✔ ~/merlict_development_kit [tidy_up_one_source_py|✚ 1…2] 
11:19 $ time ./merlict_test
___Warning___
merlict.cpp warn_about_close_frames(frame) 1236
The children of the frame 'tree' are stucked very close together.
No tree structure can be used to accelerate the photon propagation in such a scenery.
Procesing continues, but poor performance is expected.
===============================================================================
All tests passed (719042 assertions in 307 test cases)


real	0m3.973s
user	0m3.955s
sys	0m0.016s
```

Awesome!

**Comment by dneise** [comment]
Okay so I fixed the order  manually in 9d0e0c3. Now it works.

I have no idea what is up with the tests... I think we should merge #28 before me merge anything else.

File: README.md [hunk 1]
```diff
@@ -39,3 +39,29 @@ Run the unit tests in the mctracer/Tests directory to ensure your build is fine.
 cd mctracer
 ../build/merlict-test
 ```
+
+
+---
+
+# For Developers playing around with the single header + single source version.
```
Comment (relleums) [inline README.md]: thanks for adding readme

**Comment by dneise** [comment]
> I think it is not lost time because now we two already know what such a tool needs to do, and what the code should look like.

Exactly ... also ... I think ... if the alternatives are:
 - fix something which is almost working within one hour.
 - use a different tool, but research how it works within one month ... 

then these are no real alternatives ... one can be done ... until the other is ready.

File: .travis.yml [hunk 2]
```diff
@@ -26,5 +26,8 @@ script:
     - cd ..
     - ./build/merlict-test
 
-
+    - echo "trying out one_source.py"
```
Comment (relleums) [inline .travis.yml]: Very nice, this is the only way to make sure the one-source-feature is maintained in the future

**Comment by dneise** [comment]
N.B: At the moment it is very easy to test against Python 3.6 (unless one wants to use conda). I'd like to avoid using conda as long as possible...

**Comment by dneise** [comment]
Nice ... so when using the same optimization level (`-O1`) for the cmake build and the "one source build", then the unit tests take about the same time for both builds (about 2 sec), c.f.:

 - one source buid line 1143: https://travis-ci.org/cherenkov-plenoscope/merlict_development_kit/builds/517682905#L1144
 - cmake build line 1122: https://travis-ci.org/cherenkov-plenoscope/merlict_development_kit/builds/517682905#L1122

**Pull request merged**